src/engraving/rendering/dev/beamlayout.cpp

```diff
@@ -1260,6 +1272,29 @@ void BeamLayout::createBeamSegment(Beam* item, ChordRest* startCr, ChordRest* en
     }
 }
 
+int BeamLayout::computeStemShortenFactor(ChordRest* cr, Beam* beam, BeamSegment* beamSegment, int level,
+                                         ChordRest* startCr, ChordRest* endCr, std::vector<ChordRest*> stemShortenedCrs)
```

Comment:
It would be better to use a const reference (e.g. `const std::set<ChordRest*>&`) instead of creating a copy. Also, to make the intentions clear, I think you can replace the `ChordRest*` with `const ChordRest*`.